core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala

```diff
@@ -53,15 +53,20 @@ object ApplicationDescriptor {
             image: String,
             blueprint: VerifiedBlueprint,
             agentPaths: Map[String, String],
-            libraryVersion: String): ApplicationDescriptor = {
+            libraryVersion: String)(useShortK8sObjectNames: Boolean): ApplicationDescriptor = {
```

Comment:
an `apply` with 2 parameter lists is not pretty, but
- with objects it's hard to introduce configurable behavior (unless we repeatedly load a HOCON config in various objects, for instance)
- at some point the compiler expects a generic apply method `apply(?, ?, ?, ? ,?, ?, ?)`. Adding the new parameter to the other params would have confused the compiler here
- `useShortK8SObjectNames` is really configuration how to process and not a value for a data field, so a separate parameter list seems right.